apps/remote_control/lib/lexical/remote_control/api.ex

```diff
@@ -13,6 +13,37 @@ defmodule Lexical.RemoteControl.Api do
 
   require Logger
 
+  @spec maybe_schedule_compile(Project.t(), Lexical.uri()) :: :ok
+  def maybe_schedule_compile(project, triggered_file_uri) do
+    if in_rename_progress?(project) do
+      RemoteControl.call(project, Commands.Rename, :mark_saved, [triggered_file_uri])
+    else
+      schedule_compile(project, false)
+    end
+  end
+
+  @spec maybe_compile_document(Project.t(), Lexical.uri()) :: :ok
+  def maybe_compile_document(project, document) do
+    if in_rename_progress?(project) do
+      RemoteControl.call(project, Commands.Rename, :mark_changed, [document.uri])
+    else
+      compile_document(project, document)
+    end
+  end
+
+  @spec maybe_mark_closed(Project.t(), Lexical.uri()) :: :ok
+  def maybe_mark_closed(%Project{} = project, uri) do
+    if in_rename_progress?(project) do
+      RemoteControl.call(project, Commands.Rename, :mark_closed, [uri])
+    else
+      :ok
+    end
+  end
+
+  defp in_rename_progress?(%Project{} = project) do
```

Comment:
```suggestion
  defp rename_in_progress?(%Project{} = project) do
```

I'd also accept `rename_running?` to be more like `reindex`.

Reply:
Because the Build's API, such as `compile_document`, is called using `RemoteControl.call`, do you have a better suggestion?